crates/bevy_math/src/bounding/raytest2d.rs

```diff
@@ -0,0 +1,78 @@
+use super::{Aabb2d, BoundingCircle, IntersectsVolume};
+use crate::{primitives::Direction2d, Vec2};
+
+/// A raycast intersection test for 2D bounding volumes
+pub struct RayTest2d {
+    /// The origin of the ray
+    pub origin: Vec2,
+    /// The direction of the ray
+    pub dir: Direction2d,
+    /// The inverse direction of the ray
+    pub inv_dir: Vec2,
+    /// The maximum time of impact of the ray
+    pub max: f32,
+}
+
+impl RayTest2d {
+    /// Construct a [`RayTest2d`] from an origin, [`Direction2d`] and max time of impact.
+    pub fn new(origin: Vec2, dir: Direction2d, max: f32) -> Self {
```

Comment:
I think there probably should be an additional constructor (or maybe a `From` impl) from `bevy_math::ray::Ray{2,3}d` for these things

Reply:
Could even store an actual `Ray2d`/`Ray3d` in the struct instead of separate `origin` and `dir` properties. But I suppose it would make the `inv_dir` kind of isolated, so I'm not sure